### PR TITLE
PARQUET-1556 Use try-with-resource in Apache Avro tests

### PR DESCRIPTION
**Pull request by Fokko**
https://jira.apache.org/jira/browse/PARQUET-1558

### CODE REVIEW

File: parquet-avro/src/test/java/org/apache/parquet/avro/AvroTestUtil.java [hunk 1]
```diff
@@ -81,19 +81,17 @@ public static GenericRecord instance(Schema schema, Object... pairs) {
     Configuration conf = new Configuration(false);
     AvroReadSupport.setRequestedProjection(conf, schema);
     AvroReadSupport.setAvroReadSchema(conf, schema);
-    ParquetReader<D> fileReader = AvroParquetReader
-        .<D>builder(new Path(file.toString()))
-        .withDataModel(model) // reflect disables compatibility
-        .withConf(conf)
-        .build();
+    // reflect disables compatibility
```
Comment (nandorKollar) [inline parquet-avro/src/test/java/org/apache/parquet/avro/AvroTestUtil.java]: This is a duplicated comment line

Comment (Fokko) [reply parquet-avro/src/test/java/org/apache/parquet/avro/AvroTestUtil.java]: Great catch, thanks

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java [hunk 2]
```diff
@@ -89,20 +89,20 @@ public void testEmptyArray() throws Exception {
...
-    GenericData.Record record = new GenericRecordBuilder(schema)
+        .build()) {
+      GenericData.Record record = new GenericRecordBuilder(schema)
         .set("myarray", emptyArray).build();
-    writer.write(record);
-    writer.close();
+      writer.write(record);
+    }
 
     AvroParquetReader<GenericRecord> reader = new AvroParquetReader<GenericRecord>(testConf, file);
```
Comment (masayuki038) [inline parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java]: Is there any reason not to apply to AvroParquetReader?

Comment (Fokko) [reply parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java]: Good point @masayuki038

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java [hunk 3]
```diff
@@ -104,11 +104,12 @@ public void testEmptyArray() throws Exception {
       writer.write(record);
     }
 
-    AvroParquetReader<GenericRecord> reader = new AvroParquetReader<GenericRecord>(testConf, file);
-    GenericRecord nextRecord = reader.read();
+    try (AvroParquetReader<GenericRecord> reader = new AvroParquetReader<>(testConf, file)) {
```
Comment (masayuki038) [inline parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java]: I think you are fine-tuning how to put space after try for each source code. 
But isn't this different only here?

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java [hunk 4]
```diff
@@ -1,14 +1,14 @@
-/* 
+/*
```
Comment (masayuki038) [inline parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java]: (nit) Unrelated formatting-only change.

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
  * Licensed to the Apache Software Foundation (ASF) under one
  * or more contributor license agreements.  See the NOTICE file
  * distributed with this work for additional information
  * regarding copyright ownership.  The ASF licenses this file
  * to you under the Apache License, Version 2.0 (the
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
- * 
+ *
```
Comment (masayuki038) [inline parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java]: (nit) Unrelated formatting-only change.

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
  *   http://www.apache.org/licenses/LICENSE-2.0
- * 
+ *
```
Comment (masayuki038) [inline parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java]: (nit) Unrelated formatting-only change.

File: parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java [hunk 7]
```diff
@@ -339,7 +339,7 @@ private D advance() {
                   return next;
                 } catch (IOException e) {
                   throw new RuntimeException(
-                      "Failed while reading Parquet file: " + source, e);
+                    "Failed while reading Parquet file: " + source, e);
```
Comment (masayuki038) [inline parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java]: (nit) Unrelated formatting-only change.

File: parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java [hunk 8]
```diff
@@ -305,11 +305,11 @@ protected static ClassLoader loaderForPaths(List<String> paths)
...
         AvroReadSupport.setRequestedProjection(conf, projection);
         AvroReadSupport.setAvroReadSchema(conf, projection);
-        final ParquetReader<D> parquet = AvroParquetReader.<D>builder(qualifiedPath(source))
+        try(final ParquetReader<D> parquet = AvroParquetReader.<D>builder(qualifiedPath(source))
             .disableCompatibility()
             .withDataModel(GenericData.get())
             .withConf(conf)
-            .build();
+            .build()) {
         return new Iterable<D>() {
```
Comment (masayuki038) [inline parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java]: Do you not indent here?

File: parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java [hunk 9]
```diff
@@ -305,11 +305,11 @@ protected static ClassLoader loaderForPaths(List<String> paths)
         // TODO: add these to the reader builder
```
Comment (masayuki038) [inline parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java]: This file is not 'avro tests'.

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java (hunk 3 above)
Comment (Fokko) [reply parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java]: I don't think there is a clear consensus on this:
```
MacBook-Pro-van-Fokko:parquet-mr fokkodriesprong$ grep -R "try (" .
./parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java:    try (AvroParquetReader<GenericRecord> reader = new AvroParquetReader<>(testConf, file)) {
./parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java:    try (ParquetWriter<GenericData.Record> writer = AvroParquetWriter
./parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java:    try (ParquetReader<GenericRecord> reader = AvroParquetReader
./parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWriteOldListBehavior.java:    try (AvroParquetWriter<GenericRecord> writer = new AvroParquetWriter<>(file, schema)) {
./parquet-avro/src/test/java/org/apache/parquet/avro/TestArrayCompatibility.java:    try (ParquetFileReader reader = ParquetFileReader.open(new Configuration(), test)) {
./parquet-avro/src/test/java/org/apache/parquet/avro/AvroTestUtil.java:    try (ParquetReader<D> fileReader = AvroParquetReader
./parquet-avro/src/test/java/org/apache/parquet/avro/AvroTestUtil.java:    try (ParquetWriter<D> writer = AvroParquetWriter
./parquet-tools/src/test/java/org/apache/parquet/tools/read/TestSimpleRecordConverter.java:    try (
./parquet-tools/src/test/java/org/apache/parquet/tools/read/TestSimpleRecordConverter.java:    try (
./parquet-tools/src/main/java/org/apache/parquet/tools/command/ColumnIndexCommand.java:    try (ParquetFileReader reader = ParquetFileReader.open(in)) {
./parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java:    try (SeekableInput in = openSeekable(source)) {
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/ToAvroCommand.java:      try (DataFileWriter<Record> writer = w.create(projection, create(outputPath))) {
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/ConvertCommand.java:      try (ParquetWriter<Record> writer = AvroParquetWriter
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/ConvertCSVCommand.java:    try (AvroCSVReader<Record> reader = new AvroCSVReader<>(
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/ConvertCSVCommand.java:      try (ParquetWriter<Record> writer = AvroParquetWriter
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/SchemaCommand.java:        try (OutputStream out = create(outputPath)) {
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/SchemaCommand.java:    try (SeekableInput in = openSeekable(source)) {
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/ShowColumnIndexCommand.java:    try (ParquetFileReader reader = ParquetFileReader.open(in)) {
./parquet-hadoop/src/test/java/org/apache/parquet/filter2/recordlevel/PhoneBookWriter.java:    try (ParquetWriter<Group> writer = builder.build()) {
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestMultipleWriteRead.java:    try (ParquetWriter<Group> writer = ExampleParquetWriter.builder(file)
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestMultipleWriteRead.java:    try (ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), file).build()) {
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestMultipleWriteRead.java:    try (ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), file)
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetFileWriter.java:    try (ParquetFileReader reader = new ParquetFileReader(HadoopInputFile.fromPath(path, configuration),
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetWriter.java:    try (ParquetWriter<Group> writer = ExampleParquetWriter.builder(path)
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetWriter.java:    try (ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), path).build()) {
./parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/SerializationUtil.java:    try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
./parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java:    try (SeekableInputStream in = file.newStream()) {
./parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java:    try (ParquetFileReader reader = ParquetFileReader.open(file)) {
./parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ColumnIndexFilterUtils.java:      try (Formatter formatter = new Formatter()) {
./parquet-benchmarks/src/main/java/org/apache/parquet/benchmarks/FilteringBenchmarks.java:      try (ParquetWriter<Group> writer = writeConfigurator.configureBuilder(ExampleParquetWriter.builder(file)
./parquet-benchmarks/src/main/java/org/apache/parquet/benchmarks/FilteringBenchmarks.java:    try (ParquetReader<Group> reader = context.createReaderBuilder()
./parquet-benchmarks/src/main/java/org/apache/parquet/benchmarks/NestedNullWritingBenchmarks.java:    try (ParquetWriter<Group> writer = ExampleParquetWriter.builder(BLACK_HOLE)
./parquet-column/src/main/java/org/apache/parquet/internal/column/columnindex/ColumnIndexBuilder.java:      try (Formatter formatter = new Formatter()) {
./parquet-column/src/main/java/org/apache/parquet/internal/column/columnindex/OffsetIndexBuilder.java:      try (Formatter formatter = new Formatter()) {
./parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java:    try (FileWriter fw = new FileWriter(file)) {
./parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/IntBasedBitPackingGenerator.java:    try (FileWriter fw = new FileWriter(file)) {
```

From a SE point of view, I would say that either of them is fine since there is no enforcing on the rule. However, I think without space is neater. WDYT?

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java (hunk 4 above)
Comment (Fokko) [reply parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java]: This is done by my IDE. Personally, I don't like trailing whitespace, and since it doesn't affect any code, I would just remove it. We don't lose any important lineage which is the reason to not touch unrelated stuff.

File: parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java (hunk 8 above)
Comment (Fokko) [reply parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java]: My IDE didn't, do you think we should indent here?

File: parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java (hunk 9 above)
Comment (Fokko) [reply parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java]: I've reverted the change. I'll open a new PR.

File: parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java (hunk 7 above)
Comment (Fokko) [reply parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java]: Reverted the change, this was due to a different style profile that was enabled.

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java (hunk 3 above)
Comment (masayuki038) [reply parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java]: I also think that there is no policy for the whole source code about this style. I thought you were adjusting for each source code style. This issue is withdrawn.

**Comment by zivanfi** [comment]
@Fokko I just noticed after merging that this PR is not related to PARQUET-1556. Did you mean to reference a different JIRA?

**Comment by zivanfi** [comment]
I see that the correct JIRA would have been PARQUET-1558. Unfortunately I did not notice the typo before and merged this PR with a comment referencing PARQUET-1556.
